src/pages/library/DataBrowserToggler.js

```diff
@@ -1,15 +1,83 @@
+import _ from 'lodash/fp'
+import { Fragment, useState } from 'react'
 import { div, h, label, strong } from 'react-hyperscript-helpers'
-import { Link, Switch } from 'src/components/common'
+import { ButtonPrimary, IdContainer, Link, spinnerOverlay, Switch } from 'src/components/common'
+import { icon } from 'src/components/icons'
+import { TextArea, TextInput } from 'src/components/input'
+import Modal from 'src/components/Modal'
+import TooltipTrigger from 'src/components/TooltipTrigger'
+import { Ajax } from 'src/libs/ajax'
 import colors from 'src/libs/colors'
 import { isDataBrowserVisible } from 'src/libs/config'
+import { FormLabel } from 'src/libs/forms'
 import * as Nav from 'src/libs/nav'
 import { useStore } from 'src/libs/react-utils'
 import { authStore } from 'src/libs/state'
+import * as Utils from 'src/libs/utils'
 import { catalogPreviewStore } from 'src/pages/library/dataBrowser-utils'
+import validate from 'validate.js'
+
+
+const DataBrowserFeedbackModal = ({ onSuccess, onDismiss }) => {
+  const [contactEmail, setContactEmail] = useState('')
+  const [feedback, setFeedback] = useState('')
+  const [submitting, setSubmitting] = useState(false)
+
+  const submit = Utils.withBusyState(setSubmitting, async () => {
+    await Ajax().Surveys.submitForm('1FAIpQLSevEVLKiLNACAsti8k2U8EVKGHmQ4pJ8_643MfdY2lZEIusyw',
+      { 'entry.477992521': feedback, 'entry.82175827': contactEmail })
+    onSuccess()
+  })
+
+  const errors = validate({ feedback, contactEmail },
+    { feedback: { length: { minimum: 1, maximum: 2000 } }, contactEmail: contactEmail ? { email: true } : {} })
+
+  return h(Modal, {
```

Comment:
I suggest adding
```
    shouldCloseOnOverlayClick: false,
    shouldCloseOnEsc: false,
```
I've encountered feedback forms that are too quick to dismiss from a seemingly non-destructive or accidental action, at which point any feedback that I've carefully crafted is lost. This makes me then either have two pieces of feedback or, more likely, not take the time to re-type my original feedback.

Reply:
We did think about this and wanted to allow a focus loss click to dismiss the dialog, since it can be annoying when dialogs are hard to close. But if it's more typical in the terra-ui to not allow this, I can add these flags. Personally I don't like UIs where I can't use ESC to close a dialog.

Reply:
To be clear, I speak from personal experience here. I once sent some feedback about the Gmail webapp. I had fully composed it and then explored some feature about taking a screenshot, I think, and dismissing that ended up closing the entire feedback form and throwing away what I'd typed. (You'd think that I'd have learned to _always_ compose outside of web forms by now. 🙄) I did end up sending two pieces of feedback that day, but I was _very_ unhappy about it.